community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/expressions/PointFunction.scala

```diff
@@ -33,14 +33,19 @@ case class PointFunction(data: Expression) extends NullInNullOutExpression(data)
       val map = mapCreator(state.query)
       map.getOrElse("crs", CRS.WGS84.name) match {
         case CRS.Cartesian.name =>
-          val x = map("x").asInstanceOf[Double]
-          val y = map("y").asInstanceOf[Double]
+          if (!map.contains("x") || !map.contains("y")) throw new SyntaxException("A cartesian point must contain 'x' and 'y' coordinates")
+          val x = safeToDouble(map("x"))
+          val y = safeToDouble(map("y"))
           CartesianPoint(x, y)
 
         case CRS.WGS84.name =>
-          val longitude = map("longitude").asInstanceOf[Double]
-          val latitude = map("latitude").asInstanceOf[Double]
+          if (!map.contains("longitude") || !map.contains("latitude")) throw new SyntaxException("A cartesian point must contain 'x' and 'y' coordinates")
+          val longitude = safeToDouble(map("longitude"))
+          val latitude = safeToDouble(map("latitude"))
           GeographicPoint(longitude, latitude, CRS.WGS84)
+
+        case unknown => throw new SyntaxException(s"$unknown is not a supported coordinate system, supported values " +
```

Comment:
I would again say `InvalidArgumentException`. This has little to do with the syntax.